jaas/src/main/java/com/griddynamics/jagger/jaas/service/DynamicDataService.java

```diff
@@ -0,0 +1,150 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+/**
+ * Provides {@link com.griddynamics.jagger.engine.e1.services.DataService} service
+ * based on configuration described by {@link com.griddynamics.jagger.jaas.storage.model.JaggerTestDbConfig}
+ * and handles storage for {@link com.griddynamics.jagger.jaas.storage.JaggerTestDbConfigDao} entities.
+ */
+@Service
+public class DynamicDataService implements JaggerTestDbConfigDao {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicDataService.class);
+    
+    private final ExecutorService destroyerService =
+            Executors.newSingleThreadExecutor(r -> new Thread("DynamicDataServiceDestoyer"));
+    
+    private final ConcurrentMap<String, AbstractApplicationContext> dataServiceContexts = new ConcurrentHashMap<>();
+    
+    @Autowired
+    private JaggerTestDbConfigDao jaasDao;
+    
+    @Value("${jaas.data.service.cache.size}")
+    private int dataServiceCacheSize = 10;
+    
+    @Autowired
+    private JaggerTestDbConfig defaultJaggerTestDbConfig;
+    
+    @PostConstruct
+    public void init() {
+        LOGGER.info("Registering default jagger test db config: {}", defaultJaggerTestDbConfig);
+        jaasDao.createOrUpdate(defaultJaggerTestDbConfig);
+        getDataServiceFor(defaultJaggerTestDbConfig.getName());
+    }
+    
+    @PreDestroy
+    public void destroy() {
+        destroyerService.shutdown();
+    }
+    
+    protected void evictDataServiceFor(final String configName) {
+        LOGGER.info("Evicting jagger test db config with name: {}", configName);
+        AbstractApplicationContext context = dataServiceContexts.remove(configName);
+        if (context != null) {
+            destroyerService.execute(context::destroy);
+        }
+    }
+    
+    public DataService getDataServiceFor(final String configName) {
+        
+        Objects.requireNonNull(configName);
+        
+        ApplicationContext dataServiceContext = dataServiceContexts.get(configName);
+        if (Objects.isNull(dataServiceContext)) {
+            JaggerTestDbConfig config = jaasDao.read(configName);
+            if (Objects.isNull(config)) {
+                return null;
```

Comment:
Yes. I did consider this option. But since all other methods in Rest Api which work with DataService just check returned result on null I decided to go with the same approach until we redesign whole DataService